### PR TITLE
Consolidate manual changes into CloudFormation

### PR DESCRIPTION
**Pull request by housni**
The ticket: https://www.pivotaltracker.com/story/show/173948127

In order to expedite the completion of the above spike ticket, some AWS resources were manually added in.
Now that the spike is over, I wanted to consolidate the manual changes into a CloudFormation template.

[REDACTED - see comment below](https://github.com/ModusCreateOrg/aws-cf-templates/pull/6#issuecomment-2223160653)

### CODE REVIEW

File: jenkins/jenkins2-ha-agents.yaml [hunk 1]
```diff
@@ -2370,7 +2387,28 @@ Resources:
               - 'ecr:InitiateLayerUpload'
               - 'ecr:PutImage'
               - 'ecr:UploadLayerPart'
-
+  DeploymentDockerRegistry:
+    Type: AWS::ECR::Repository
+    DeletionPolicy: Retain
+    Properties:
+      RepositoryName: iloan/deploy/ansible
```
Comment (seanmccleary) [inline jenkins/jenkins2-ha-agents.yaml]: This line does not look generic. Is it correct?

**Review by seanmccleary — APPROVED** [review]
Approved because I'm not entirely familiar with this repo and its best practices, but please do double-check the comment that I left.

File: jenkins/jenkins2-ha-agents.yaml [hunk 2]
```diff
@@ -530,6 +534,19 @@ Resources:
       LoadBalancerArn: !Ref MasterELB
       Port: 80
       Protocol: HTTP
+  MasterELBListenerHTTPS:
```
Comment (seanmccleary) [inline jenkins/jenkins2-ha-agents.yaml:544]: If we've altered the security group to only allow ingress on port 443 and not on port 80 any longer, how come we still need the ELB listener on port 80 above?

File: jenkins/jenkins2-ha-agents.yaml [hunk 3]
```diff
@@ -530,6 +534,19 @@ Resources:
...
+    Type: 'AWS::ElasticLoadBalancingV2::Listener'
+    DeletionPolicy: Retain
```
Comment (seanmccleary) [inline jenkins/jenkins2-ha-agents.yaml:546]: Just curious: Why retain this if the stack is torn down?

File: jenkins/jenkins2-ha-agents.yaml [hunk 4]
```diff
@@ -2370,7 +2387,28 @@ Resources:
               - 'ecr:InitiateLayerUpload'
               - 'ecr:PutImage'
               - 'ecr:UploadLayerPart'
-
+  DeploymentDockerRegistry:
```
Comment (seanmccleary) [inline jenkins/jenkins2-ha-agents.yaml]: @housni you mentioned on Slack that there may need to be a different ECR for each github repo.  Why is that?  I think I'm still missing a piece of the puzzle here.

**Comment by obscurerichard** [comment]
I'm editing the description of this issue to eliminate the IP addresses and AWS IDs and notations there to protect the privacy and confidentiality of the owners. They are _not_ in the code itself. I saved a copy locally in case we need it, but it's almost 4 years later.

**Pull request merged**